Code/src/main/java/nl/utwente/group10/ui/components/Block.java

```diff
@@ -19,4 +26,23 @@ public void setSelected(boolean selectedState) {
 		//TODO If another object is selected then deselect it first!!
 		isSelected = selectedState;
 	}
+	
+	@Override
+	public void handleCustomEvent(UIEvent event) {
+		EventType<UIEvent> eventType = (EventType<UIEvent>) event
```

Comment:
Is this cast necessary? I'm not sure, but you might get away with just calling this an `EventType` instead of an `EventType<UIEvent>`.